sympy/parsing/latex/lark/transformer.py

```diff
@@ -125,19 +135,45 @@ def gte(self, tokens):
         return sympy.Ge(tokens[0], tokens[2])
 
     def add(self, tokens):
-        return sympy.Add(tokens[0], tokens[2])
+        if len(tokens) == 2: # +a
+            return tokens[1]
+        if len(tokens) == 3: # a + b
+            lh = tokens[0]
+            rh = tokens[2]
+
+            if self._obj_is_sympy_Matrix(lh) or self._obj_is_sympy_Matrix(rh):
+                return sympy.MatAdd(lh, rh)
+
+            return sympy.Add(lh, rh)
 
     def sub(self, tokens):
-        if len(tokens) == 2:
-            return -tokens[1]
-        elif len(tokens) == 3:
-            return sympy.Add(tokens[0], -tokens[2])
+        if len(tokens) == 2: # -a
```

Comment:
These short comments next to the `len(tokens) == ...` lines are a really good idea. Nice work!

Reply:
Thank you!